coordinator/src/node.rs

```diff
@@ -174,11 +196,45 @@ impl Node {
                     None => bail!("Failed to find open position : {}", trade_params.pubkey),
                 };
 
-                self.close_position(&position, closing_price, channel_id, fee_payment_hash)
-                    .await?
+                if let Err(e) = self
+                    .close_position(&position, closing_price, channel_id, fee_payment_hash)
+                    .await
+                {
+                    orders::set_order_state(
+                        &mut connection,
+                        trade_params.filled_with.order_id,
+                        OrderState::Failed,
+                    )?;
+
+                    matches::set_match_state(
+                        &mut connection,
+                        trade_params.filled_with.order_id,
+                        MatchState::Failed,
+                    )?;
```

Comment:
Same comment as above. It's my understanding that these db updates should run atomically.